src/pg_to_evalscript/evalscript.py

```diff
@@ -92,12 +92,21 @@ def write_update_output(self):
     }});
 }}"""
 
+    def write_output_variable(self):
+        if len(self.nodes) == 0:
```

Comment:
`self.nodes` is a list of instances of `Node` class (https://github.com/openEOPlatform/openeo-pg-evalscript-converter/blob/eebaefe694d48f7e28c6233daf1ed529a226a86b/src/pg_to_evalscript/node.py). 

It is generated here: https://github.com/openEOPlatform/openeo-pg-evalscript-converter/blob/eebaefe694d48f7e28c6233daf1ed529a226a86b/src/pg_to_evalscript/conversion.py#L127-L173

So this method iterates over the "nodes" in the process graph and creates `Node` objects from them. If the node is empty (it doesn't have `process_id` or `arguments` defined) it will indeed raise an error. But the process graphs aren't being validated anywhere. 

Perhaps this could be added, I guess using the `openeo-pg-parser-python`. Although for our use case we already validate the process graphs when the request is made.